### PR TITLE
Add explorer tooltips for proposal IDs and validator addresses

### PR DESCRIPTION
**Pull request by rmeissner**
## Summary

This PR adds two targeted tooltip enhancements to the explorer to surface existing but hidden data:

1. **Proposal details** — Display Signature ID and Group ID in the attestation status section with copy buttons
2. **Validator labels** — Show full checksummed addresses in native browser tooltips when hovering over validator names

## Key Changes

- **SafeTxAttestationStatus.tsx**: Added two new labeled rows displaying `Signature ID` and `Group ID` using existing `InlineHash` and `CopyButton` components. Both values are already fetched but were previously unused in the render output.

- **ValidatorList.tsx**: Refactored the validator list render from a simple joined string to JSX elements with per-item `title` attributes. Each validator name is now wrapped in a `<span>` with the full address as the tooltip, enabling native browser tooltips on hover. The component API and all call sites remain unchanged.

- **ValidatorList.test.tsx**: Added comprehensive test coverage for the refactored render logic, including validation of title attributes, emoji suffixes, sort order, and edge cases.

## Implementation Details

- Uses native HTML `title` attributes for tooltips (consistent with existing `Badge` and `NetworkBadge` patterns) rather than introducing a third-party tooltip library
- Maintains existing layout and styling patterns (`md:flex md:justify-between`)
- No new components created; reuses existing `InlineHash` and `CopyButton` components
- No changes to component APIs or call sites
- Preserves loading skeleton behavior in `SafeTxAttestationStatus`

https://claude.ai/code/session_01R8A6eRbMRzZy4bCfgunkNC

### CODE REVIEW

File: explorer/src/components/common/ValidatorList.test.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	it("shows the validator address in a popover when clicking a validator label", () => {
+		render(<ValidatorList all={[ADDR_A]} active={[ADDR_A]} mapInfo={mapInfo} completed={false} />);
+		fireEvent.click(screen.getByText("Alice ✅"));
+		expect(screen.getByText(ADDR_A)).toBeTruthy();
+	});
+
+	it("shows a copy button in the popover when clicking a validator label", () => {
+		render(<ValidatorList all={[ADDR_A]} active={[ADDR_A]} mapInfo={mapInfo} completed={false} />);
+		fireEvent.click(screen.getByText("Alice ✅"));
+		expect(screen.getByRole("button", { name: "Copy to clipboard" })).toBeTruthy();
```
Comment (rmeissner) [inline explorer/src/components/common/ValidatorList.test.tsx]: This can be combined with the test before. The whole pop up content can be validated in one test

**Review by nlordell — APPROVED** [review]
